### DISCUSSION TITLE
Add an argument to kill (itself) nuclei process based on % of scan done

### DISCUSSION THREAD

**Discussion by ghost** [projectdiscovery/nuclei#4356] [General]
### Please describe your feature request:

Like the title says, an argument to kill the nuclei process based on % of scan. You know when you do -stats you see this for example:
[5:25:09] | Templates: 4711 | Hosts: 224 | RPS: 75 | Matched: 107 | Errors: 1042174 | Requests: 1482183/1723232 (86%)

I want to be able to kill (itself) the nuclei process when the scan reaches 86% or 89%. A % I wish.

### Describe the use case of this feature:

Personally, I do parallel scanning and it's probably my code that causes the bottlenecks but nuclei just starts hanging, I stop seeing the results for -stats (probably because I don't set -si 1 but by default its 5 so it should still give the stats output). 

Adding the argument I'm talking about could be beneficial in my opinion because I can just end the nuclei process itself at % of the scan finished.

I hope this makes sense.

**Comment by sec13b** [comment]
how you got RPS: 75 ?
 | Templates: 1 | Hosts: 461276 | RPS: 2 | Matched: 0 | Errors: 3866 | Requests: 3923/461276 (0%)

**Comment by tarunKoyalwar** [comment]
Hii @ghost  , there was a bug in nuclei before v3 so if multiple nuclei instances were running there was possibility of stats being stopped etc . but it should be stable now . 

about stopping nuclei process if it hung , you can use `-hang-monitor` flag . when this flag is enabled nuclei will check if nuclei is stopped/hung and if it was detected that nuclei will exit automatically with a stacktrace / description of where nuclei hung/frozen

```console
$  nuclei -h hang-monitor                                                             1 ↵
Nuclei is a fast, template based vulnerability scanner focusing
on extensive configurability, massive extensibility and ease of use.

Usage:
  nuclei [flags]

Flags:
   -hm, -hang-monitor  enable nuclei hang monitoring
```

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
